arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/halt/SelfdestructSection.java

```diff
@@ -90,7 +90,7 @@ public SelfdestructSection(Hub hub) {
 
     recipientAccountBefore =
         selfDestructTargetsItself
-            ? selfdestructorAccountAfter.deepCopy()
+            ? selfdestructorAccountBefore.deepCopy()
```

Comment:
AFAICT this makes no sense in terms of the HUB constraints. And since we aren't testing the HUB constraints what was solved by this ?

Reply:
@OlivierBBB `selfdestructorAccountAfter` was not initialised at that point of the code.

Reply:
I switched back to `selfdestructorAccountAfter` and initialised it before it is used @OlivierBBB.

Reply:
TODO: @lorenzogentile404 test 1104b is failing. 
If after updating this branch it still fails, create a new issue documenting:

```
- transaction number
- hubStamp
-contextNumber
- addresses
- type of call (CALL, DELEGATECALL, ...)
- stamp at revert
- depth of call ...
```

Reply:
It may make sense to
- in the constructor
  - capture the `senderAccountBefore` in all cases 
  - capture the `recipientAccountBefore` the `SELFDESTRUCT` only in case `selfDestructTargetsItself == false`
- in `traceContextExit`
  - capture the `senderAccountAfter` in all cases at `traceContextExit`
  - capture the `recipientAccountAfter` in all cases at `traceContextExit`

**Note.** I propose `traceContextExit` but it could be that `traceContextReEntry` is more adequate for the task. I'm not when the account change (to the balances) takes place for `SELFDESTRUCT` and I'm also not sure what would happen with `traceContextExit` if the `SELFDESTRUCT` were to take place in the `ROOT` context. This is why I think `traceContextExit` may be the more appropriate choice.